### DISCUSSION TITLE
Partial System.IO access

### DISCUSSION THREAD

**Discussion by TheKing349** [Hello-Meow/ModTool#24] [Q&A] [answered]
Hi! I was just wondering if there is a way to only allow one specific directory to be read/writable? I want the modder to be able to have save files within their mod, but to do that I would need to not restrict System.IO. Obviously, this is not secure by any means, so only allowing one directory would be beneficial. I have no idea if this idea has any merit or it is even possible at all, just thought I would ask.

**Comment by Hello-Meow** [comment, marked as answer]
You can do this by sharing a class from your game that lets them load or save something. You can do this by creating a script and assembly definition and adding them both to the exporter in the shared assets window.

**Reply by TheKing349** [reply]
Ohhhh that makes sense! I was trying to modify System.IO directly to only allow that, but synchronizing the saving and loading system would prove to be more useful. Thank you!!
